src/components/full-screen-popover/FullScreenPopover.js

```diff
@@ -0,0 +1,143 @@
+// @flow
+import * as React from 'react';
+import uniqueId from 'lodash/uniqueId';
+
+import classNames from 'classnames';
+import { Overlay } from '../flyout';
+import { KEYS } from '../../constants';
+import OverlayHeader from './OverlayHeader';
+import ScrollWrapper from '../scroll-wrapper';
+import './styles/FullScreenPopover.scss';
+
+export type FullScreenPopoverProps = {
+    /** Components to render in the overlay */
+    children?: React.Node,
+    /** Set className to the overlay wrapper */
+    className?: string,
+    /** Content nested inside of the popover header */
+    header?: React.Element<any>,
+    /** Fires this callback when overlay is closed */
+    onClose?: Function,
+    /** CSS selector that identifies clickable elements that closes the overlay */
+    onCloseCssSelector?: string,
+    /** Fires this callback when overlay is opened */
+    onOpen?: Function,
+    /** Custom button that toggles the overlay */
+    popoverButton: React.Element<any>,
+};
+
+type Props = FullScreenPopoverProps;
+
+const FullScreenPopover = (props: Props) => {
```

Comment:
The concept here seems similar to the existing Flyout and/or Modal components. Can we reuse that by adding an `isFullscreen` option to it?